compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -137,6 +137,9 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
     ) -> LexicalRegionResolutions<'tcx> {
         let mut var_data = self.construct_var_data();
 
+        self.data.constraints.sort_by_key(|(constraint, _)| *constraint);
+        self.data.constraints.dedup_by_key(|(constraint, _)| *constraint);
```

Comment:
since they get deduped anyway the order of duplicates can't matter. Have you checked how an unstable sort performs? Or is the comparison expensive.

Reply:
Using unstable sort would result in the `SubregionOrigin` being picked non-deterministically.

Reply:
Exactly! We need to report the `SubregionOrigin` of the constraint with the least index in order to preserve the previous behavior.